Managers/UInputMan.cpp

```diff
@@ -837,10 +838,15 @@ namespace RTE {
 			return;
 		}
 
-		if (g_InActivity) {
+		if (g_InActivity) {	
+
 			if (AnyStartPress(false)) {
-				g_ActivityMan.PauseActivity();
-				return;
+				GameActivity* currentGameActivity = dynamic_cast<GameActivity*>(g_ActivityMan.GetActivity());
```

Comment:
One final little comment that I didn't think about before - now that all the logic for this has been moved to BuyMenuGUI, you can just inline this variable into the if check, i.e. 
```cpp
if (!dynamic_cast<GameActivity*>(g_ActivityMan.GetActivity())->IsBuyGUIVisible(-1)) {
	g_ActivityMan.PauseActivity();
	return;
}
```

Reply:
In the general case for stuff like this, if it's only single use and clear as to what it does are inline calls like the example you've given acceptable?

Reply:
Yeah totally. Make variables if
1. You're reusing it in multiple places, cause you get some obvious efficiency benefits (though the compiler might figure this one out anyway)
2. It helps with readability, e.g. some function calls might be very long with everything inline. You'll just have to use your judgement on this